### PR TITLE
IPROTO-230 Fix MarshallerByteCodeGenerator on JDK 17

### PR DESCRIPTION
**Pull request by tristantarrant**
https://issues.redhat.com/browse/IPROTO-230

### CODE REVIEW

File: core/src/main/java/org/infinispan/protostream/annotations/impl/types/ReflectionTypeFactory.java [hunk 1]
```diff
@@ -1,6 +1,7 @@
 package org.infinispan.protostream.annotations.impl.types;
 
 import java.lang.annotation.Annotation;
+import java.lang.invoke.MethodHandles;
```
Comment (pruivo) [inline core/src/main/java/org/infinispan/protostream/annotations/impl/types/ReflectionTypeFactory.java:4]: these changes can be reverted IMO

Comment (tristantarrant) [reply core/src/main/java/org/infinispan/protostream/annotations/impl/types/ReflectionTypeFactory.java]: oops, fixed

**Comment by pruivo** [comment]
got the following error
```
[ERROR] Failures: 
[ERROR]   AnnotationProcessorCompilationTest.testAdapter:133 Expected 0 warnings, but found the following 1 warnings:
Integer(int) in java.lang.Integer has been deprecated and marked for removal
```

```
Apache Maven 3.8.1 (05c21c65bdfed0f71a2f2ada8b84da59348c4c5d)
Maven home: /home/pedro/.opt/java/maven
Java version: 17.0.1, vendor: Eclipse Adoptium, runtime: /home/pedro/.opt/java/jdk-17.0.1+12
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "5.16.14-1-manjaro", arch: "amd64", family: "unix"
```

It looks like the generator needs to be updated. Investigating... 
```
/SOURCE_OUTPUT/test_marshall_externals/AddressAdapter$___Marshaller_ce46b223cf7571fb6dc96c1c4cdd0435322f78d395b675f266ac6e23af7d306b.java:54: warning: [removal] Integer(int) in java.lang.Integer has been deprecated and marked for removal
               __v$3 = new java.lang.Integer($in.readInt32());
                       ^
```

**Comment by pruivo** [comment]
`org.infinispan.protostream.annotations.impl.AbstractMarshallerCodeGenerator` line 875

**Comment by tristantarrant** [comment]
I have fixed the deprecations

**Comment by pruivo** [comment]
merged! thanks @tristantarrant !

**Pull request merged**